agent/connect/ca/provider_vault.go

```diff
@@ -637,3 +695,64 @@ func ParseVaultCAConfig(raw map[string]interface{}) (*structs.VaultCAProviderCon
 
 	return &config, nil
 }
+
+func vaultLogin(client *vaultapi.Client, authMethod *structs.VaultAuthMethod) (*vaultapi.Secret, error) {
+	loginPath, err := configureVaultAuthMethod(authMethod)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Logical().Write(loginPath, authMethod.Params)
+	if err != nil {
+		return nil, err
+	}
+	if resp == nil || resp.Auth == nil || resp.Auth.ClientToken == "" {
+		return nil, fmt.Errorf("login response did not return client token")
+	}
```

Comment:
curious if we could surface any err codes from vault api if present

Reply:
🤔 In this case there aren't any and this is taken from the [code example](https://www.vaultproject.io/docs/auth/kubernetes#code-example) in the docs.

Reply:
Oh I see, `resp` there is a `Secret*`.

Hmm that makes sense!